domain/application/state/storage.go

```diff
@@ -34,6 +34,25 @@ WHERE  charm_uuid = $applicationDetails.charm_uuid
 		return errors.Capture(err)
 	}
 
+	// Storage is either a storage type or a pool name.
+	// Get a mapping of pool name to pool UUID for all pools.
+	// The number of pools is always small - we'll limit the size
+	// of the result set just in case. If there's more than 200 there's
+	// much bigger problems in play.
+	storageQuery, err := st.Prepare(`SELECT &storagePool.* FROM storage_pool LIMIT 200`, storagePool{})
```

Comment:
Don't do it like this. Just select uuid and name with an `IN` clause, and get the disjoint in memory.

Although this will likely be correct for every case, it's circumstantial. Think algebraically.

Reply:
changed to use IN clause